webapp/packages/core/src/root/PermissionsService.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * cloudbeaver - Cloud Database Manager
+ * Copyright (C) 2020 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { injectable } from '@dbeaver/core/di';
+import { GraphQLService, CachedResource } from '@dbeaver/core/sdk';
+
+type PermissionsMetadata = {
+  loaded: boolean;
+}
+
+@injectable()
+export class PermissionsService {
+  private permissions = new CachedResource(
+    new Map(),
+    this.refreshAsync.bind(this),
+    (_, { loaded }) => loaded
+  );
+
+  constructor(
+    private graphQLService: GraphQLService,
+  ) { }
+
+  has(id: string): boolean {
+    return this.permissions.data.has(id);
+  }
+
+  async hasAsync(id: string): Promise<boolean> {
```

Comment:
never required. Permissions will be loaded on application bootstrap. there will be no cases in normal workflow when there are no permissions

Reply:
fixed